tests/test_alpha_3.py

```diff
@@ -1,85 +1,280 @@
-import unittest 
 import os 
+import random
 import sys 
+import unittest 
+
+import numpy as np
+import pennylane as qml
+import torch
+from parameterized import parameterized_class
+
+
+# The two lines below will be removed when converting the library to a package.
 current_path = os.path.dirname(os.path.abspath(__file__))
-sys.path.append(current_path + "/../neasqc_wp61/models/quantum/alpha/module/")
-from alpha_3_trainer import Alpha3Trainer
-from alpha_3_model import Alpha3Model
-from torch.utils.data import DataLoader
-from utils import seed_everything, preprocess_train_test_dataset_for_alpha_3, BertEmbeddingDataset
+sys.path.append(current_path + "/../neasqc_wp61/models/quantum/")
+sys.path.append(current_path + "/../neasqc_wp61/models/quantum/alpha_3/")
+import circuit as circ
+from alpha_3 import Alpha3 as alpha_3
+from utils import get_labels_one_hot_encoding
 
-class TestAlpha3(unittest.TestCase):
+test_args = {
```

Comment:
🤔 confused by this...
Can I get details of what it's all about?

Reply:
From what I understood from my chats with Pablo S. about parametrised tests using 'parameterized' is that in his approach, he only uses the parameterized package to define the parametrisation and does not use any subTests. This means that no setUp or setUpClass is required. 

He defines these variables test_args which then inform the method set_up_parameters on how many values of each variable to randomly create for testing, what ranges these values should be in, etc. 

set_up_parameters will then output a list consisting of a number of lists with the different values to be used in testing. Then one must , in the same order, define a list of names for each of these lists of values. parameterized then uses this list of names and the associated list of lists of values to create the parameterized classes for the tests.

He explained this to me as the way the parameterized pipeline was explained in the examples he saw online and in the documentation of the package.